src/tests/JIT/opt/Vectorization/StringEquals/StringEquals.cs

```diff
@@ -0,0 +1,364 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+public class StringEquals
+{
+    public static int Main()
+    {
+        int testCount = 0;
+        foreach (var method in typeof(Tests).GetMethods())
```

Comment:
Another way (that avoids reflection and repeating Equals_# methods) could be to make `ValidateEquals` methods public and dup the string instances:

```c#
         int testCount = 0;
         foreach (string testStr in Tests.s_TestData)
         {
             testCount++;
             string newStr = new string(testStr.ToCharArray());
             Tests.ValidateEquals(testStr == newStr, testStr, newStr);
         }
```

Reply:
Do you mean to use SG/T4 generation? Because I do need explicit literals in method to test

Reply:
> explicit literals

Ah, ok. 👍
(SG/T4 would pull in more infra code here than necessary. I was thinking if intention is to test other instance of same string, then we can clone it via char[] overload and reduce LOC)